### PR TITLE
feat(python): Added tqdm handler for `Engine.update` progress bar.

### PR DESCRIPTION
**Pull request by schmidmt**
The API for the Python binding did not change outside the `lace.core` module.

This allows for a progress bar to be called and updated like `Lace::UpdateHandler` in `lace.core.update`. Here, the default is to use `tqdm.auto`.

This addresses #173.

### CODE REVIEW

File: pylace/src/lib.rs [hunk 1]
```diff
@@ -1027,19 +1029,20 @@ impl CoreEngine {
             checkpoint=None,
             transitions=None,
             save_path=None,
-            quiet=false,
+            update_handler=None,
         )
     )]
     fn update(
         &mut self,
+        py: Python<'_>,
```
Comment (Swandog) [inline pylace/src/lib.rs:1037]: Is there a need for the `py` object if there is no `update_handler`? Is it still worth it to `allow_threads`?

File: pylace/src/update_handler.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        map
+    }};
+}
+
+macro_rules! call_pyhandler_noret {
+    ($self: ident, $func_name: tt, $($key: tt : $val: expr),* $(,)?) => {{
+        let handler = $self
+            .handler
+            .lock()
+            .expect("Should be able to get a lock for the PyUpdateHandler");
```
Comment (Swandog) [inline pylace/src/update_handler.rs:54]: I'm not 100% sure what happens if you panic inside a Rayon thread. Will it do what we want (i.e., abort all execution)?

**Review by Swandog — APPROVED** [review]
Had a couple of questions, but I legitimately don't know if they are problems, so I have no concerns with merging this. LGTM!

File: pylace/src/lib.rs (hunk 1 above)
Comment (schmidmt) [reply pylace/src/lib.rs:1037]: Yeah, during the update step, it allows other threads to run as it yields the GIL. This doesn't change things for us but it does make the code more usable to other users.

File: pylace/src/update_handler.rs (hunk 2 above)
Comment (schmidmt) [reply pylace/src/update_handler.rs:54]: Yes, it should panic and bring down the whole process via the standard panic handler.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a33a4109f7897af06cac6e083f8cf76a

**Review by Swandog — APPROVED** [review]

**Pull request merged**